libnavigation-core/src/main/java/com/mapbox/navigation/core/RoutesSetCallback.kt

```diff
@@ -1,107 +1,161 @@
 package com.mapbox.navigation.core
 
+import androidx.annotation.IntDef
 import com.mapbox.navigation.base.route.NavigationRoute
 
 /**
  * Interface definition for a callback that gets notified whenever routes
  * passed to [MapboxNavigation.setNavigationRoutes] are set or produce an error and ignored.
  */
-interface RoutesSetCallback {
+fun interface RoutesSetCallback {
 
     /**
      * Invoked whenever the routes passed to [MapboxNavigation.setNavigationRoutes]
-     * are successfully set.
+     * are successfully set produce an error and are ignored.
      *
-     * @param result [RoutesSetCallbackSuccess] object that describes the set routes
+     * @param result [RoutesSetResult] object that describes the result of
+     *   [MapboxNavigation.setNavigationRoutes] invocation
      */
-    fun onRoutesSetResult(result: RoutesSetCallbackSuccess)
-
-    /**
-     * Invoked whenever the routes passed to [MapboxNavigation.setNavigationRoutes]
-     * produce an error and are ignored.
-     *
-     * @param result [RoutesSetCallbackError] object that describes the failure.
-     */
-    fun onRoutesSetError(result: RoutesSetCallbackError)
+    fun onRoutesSet(result: RoutesSetResult)
 }
 
 /**
- * Represents the result of setting routes. See [RoutesSetCallback.onRoutesSetResult].
+ * Represents the result of setting routes. See [RoutesSetCallback.onRoutesSet].
  *
- * @param routes List of routes that were successfully set.
+ * @param status status of the [MapboxNavigation.setNavigationRoutes] invocation.
+ *   See [RoutesSetStatus] for possible values.
+ * @param passedRoutes list of routes that were passed to [MapboxNavigation.setNavigationRoutes].
+ * @param primaryRoute status of setting the primary route to the navigator.
+ *   If null, no primary route was passed. See [RouteStatus] for more info.
+ * @param acceptedAlternatives list of alternative routes that were accepted by the navigator.
+ *   See [RouteStatus] for more info.
+ * @param ignoredAlternatives list of alternative routes that were ignored by the navigator.
+ *   See [RouteStatus] for more info.
  */
-class RoutesSetCallbackSuccess internal constructor(
-    val routes: List<NavigationRoute>
+class RoutesSetResult internal constructor(
+    @Status val status: Int,
+    val passedRoutes: List<NavigationRoute>,
+    val primaryRoute: RouteStatus?,
+    val acceptedAlternatives: List<RouteStatus>,
+    val ignoredAlternatives: List<RouteStatus>,
 ) {
```

Comment:
```suggestion
/**
* Result when the primary route has been successfully set and we can begin Active Guidance.
*/
class RoutesSetSuccess internal constructor(
    val providedRoutes: List<NavigationRoute>,
    val setPrimaryRoute: NavigationRoute,
    val setAlternatives: List<NavigationRoute>,
    val ignoredAlternatives: List<RouteSetError>,
) {
}

/**
* Result when a route failed to be set.
*/
class RoutesSetError internal constructor(
    val route: NavigationRoute,
    val message: String
) {
}
```

Reply:
What should be passed as a route to `RoutesSetError`? The primary route? There's no guarantee it will be present. As I mentioned, we can invoke `setNavigationRoutes` with an empty list. NN will return success, not error. But should we rely on that? I'd prefer not to. So I say we either make route nullable or introduce a third class: success, error (where we have the message and maybe just the providedRoutes list) and sth like `IgnoredAlternative`, which is current `RoutesSetError`, WDYT?

Reply:
> if for example there's one callback for every invocation that reports bad routes to some analytics , it won't be able to extract them from RoutesSetError

It would, the consumer can pair the route with the error message - the initial `result.error` would always refer to the primary route they were trying to set (which they have the access to). The `result.value.ignoredAlternatives` would refer to the potentially failed alternatives and round-trips the routes.

> several equal alternatives

Hmm, that's kind of a mistake in itself. We could also try:
```kotlin
class RoutesSetSuccess internal constructor(
    val ignoredAlternatives: Map<Int, RouteSetError>,
) {
}
```
Where the key is `NavigationRoute#id` which would again allow developers to match it with the `NavigationRoute` refs they tried to set and report to analytics if needed.